### PR TITLE
[Example] Added gRPC C++ cancellation example

### PR DESCRIPTION
**Pull request by veblush**

### CODE REVIEW

File: examples/cpp/cancellation/client.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
```
Comment (yashykt) [inline examples/cpp/cancellation/client.cc:3]: fix copyright year

File: examples/cpp/cancellation/server.cc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+using grpc::Server;
+using grpc::ServerBidiReactor;
+using grpc::ServerBuilder;
+using grpc::Status;
+using keyvaluestore::KeyValueStore;
+using keyvaluestore::Request;
+using keyvaluestore::Response;
+
+// Logic behind the server's behavior.
+class KeyValueStoreServiceImpl final : public KeyValueStore::CallbackService {
```
Comment (yashykt) [inline examples/cpp/cancellation/server.cc:48]: This is not a key value store. I think you should add a method to helloworld.proto to have a bidi streaming RPC

**Review by yashykt — APPROVED** [review]

File: examples/cpp/cancellation/server.cc [hunk 3]
```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
```
Comment (yashykt) [inline examples/cpp/cancellation/server.cc:3]: fix copyright year here as well

File: examples/cpp/cancellation/client.cc (hunk 1 above)
Comment (veblush) [reply examples/cpp/cancellation/client.cc]: Done

File: examples/cpp/cancellation/server.cc (hunk 2 above)
Comment (veblush) [reply examples/cpp/cancellation/server.cc]: Done

File: examples/cpp/cancellation/server.cc (hunk 3 above)
Comment (veblush) [reply examples/cpp/cancellation/server.cc]: Done

**Pull request merged**